metricproviders/datadog/datadog.go

```diff
@@ -173,6 +250,56 @@ func (p *Provider) parseResponse(metric v1alpha1.Metric, response *http.Response
 	return strconv.FormatFloat(value, 'f', -1, 64), status, err
 }
 
+func (p *Provider) parseResponseV2(metric v1alpha1.Metric, response *http.Response) (string, v1alpha1.AnalysisPhase, error) {
+
+	bodyBytes, err := io.ReadAll(response.Body)
+
+	if err != nil {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("Received no bytes in response: %v", err)
+	}
+
+	if response.StatusCode == http.StatusForbidden || response.StatusCode == http.StatusUnauthorized {
+		return "", v1alpha1.AnalysisPhaseError, fmt.Errorf("received authentication error response code: %v %s", response.StatusCode, string(bodyBytes))
```

Comment:
Will this log any credentials?

Reply:
This shouldn't as this is just the same logic for parsing as in v1: https://github.com/argoproj/argo-rollouts/blob/master/metricproviders/datadog/datadog.go#L141